common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -279,6 +281,61 @@ def verify_current_content_visibility(cls, date, hide_after_date):
             datetime.now(UTC) < date
         )
 
+    def gate_sequence_if_it_is_a_timed_exam_and_contains_content_type_gated_problems(self):
+        """
+        Problem:
+        Content type gating for FBE (Feature Based Enrollments) previously only gated individual blocks.
+        This was an issue because audit learners could start a timed exam
+        and then be unable to complete it because the graded content would be gated.
+        Even if they later upgraded, they could still be unable to complete the exam
+        because the timer could have expired.
+
+        Solution:
+        Gate the entire sequence when we think the above problem can occur.
+
+        If:
+        1. This sequence is a timed exam
+        2. And this sequence contains problems which this user cannot load due to content type gating
+        Then:
+        We will gate access to the entire sequence.
+        Otherwise, learners would have the ability to start their timer for an exam,
+        but then not have the ability to complete it.
+
+        We are displaying the gating fragment within the sequence, as is done for gating for prereqs,
+        rather than content type gating the entire sequence because that would remove the next/previous navigation.
+
+        This functionality still needs to be replicated in the frontend-app-learning courseware MFE
+        The ticket to track this is https://openedx.atlassian.net/browse/REV-1220
+        Note that this will break compatability with using sequences outside of edx-platform
+        but we are ok with this for now
+        """
+        if not self.is_time_limited:
+            return
```

Comment:
Should this also set `self.gated_sequence_fragment = None` so that access is not blocked?

Reply:
I don't think it would actually be necessary here but i can add it here to be safe